packages/web-config-server/src/apiV1/utils/findAccessibleGroupedMapOverlays.js

```diff
@@ -184,11 +223,12 @@ export const findAccessibleGroupedMapOverlays = async (models, accessibleMapOver
 
     if (isNonEmptyMapOverlayGroup) {
       accessibleRelations.push(worldRelationById[groupId]);
-      accessibleOverlayGroups.push({
+      const accessibleOverlayGroup = integrateMapOverlayReference({
+        mapOverlays: nestedMapOverlayGroups,
```

Comment:
`nestedMapOverlayGroups` is probably a list of map overlays in Laos EOC because we only have 1 level of nesting there. But if we do this in multi nested level groups like Laos Schools, `nestedMapOverlayGroups` will be a list of over lay groups I believe. Example screenshot here: https://github.com/beyondessential/tupaia-backlog/issues/1788#issuecomment-784827175

In addition, the current code also supports having group and overlays in the same group (no projects use this at the moment, but can be in the future). Example screenshot is here: https://github.com/beyondessential/tupaia-backlog/issues/194#issuecomment-679847182. So `nestedMapOverlayGroups` can actually contain both map overlays and overlay groups.

Fortunately I think you probably don't need to rewrite too much? (I might be a bit too naive). Perhaps the below code should be enough? If so the same has to be done in the other place that you call the same function

```suggestion
      const accessibleOverlayGroup = integrateMapOverlayReference({
        mapOverlays: nestedMapOverlayGroups, // do filter for map overlays here
```

It'd be great if you can test locally if it works with overlays in multi nested groups in Laos Schools as well.

Reply:
Thanks Chris for these sharing! I thought this new recursive function can deal with having group and overlays in the same group, but I didn't test it to be honest. 

I guess option 2 will have conflict when two sub groups has different source, so I agree options 1 makes sense to me and is easier to implement. I will go and implement this.